### PR TITLE
[Stardog] Only run postinstall if replicas > 0

### PR DESCRIPTION
**Pull request by chamer81**
#### What this PR does / why we need it:
This prevents the postinstall job from running when the chart is installed with replicaCount equal to zero. The postinstall job requires at least one running pod in order to terminate successfully, so setting the replica count to zero via gitops would always lead to a fail state without this change.
#### Checklist

- [x] [DCO](https://github.com/appuio/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR contains starts with chart name e.g. `[chart]`

### CODE REVIEW

**Comment by srueg** [comment]
Can you please add a test in the `tests/` folder to verify the behavior?
And I think you can target the `master` branch with this PR.

**Comment by chamer81** [comment]
I have added the unit tests. But since I created this branch from the stardog/update-to-v7 branch, I can't merge it into master without bringing along all of those changes -- unless the idea is to cherry-pick the commits to a new branch from master. @srueg what do you suggest?

**Comment by srueg** [comment]
Yes, you can reset this branch to `master` and then cherry-pick the two commits for this change 👍

File: stardog/test/postinstall_test.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	options := &helm.Options{
+		SetValues: map[string]string{
+			"stardog.adminPassword":     adminPassword,
+			"replicaCount":	"0",
+		},
+	}
+
+	output, err := helm.RenderTemplateE(t, options, helmChartPath, releaseName, tplPostInstallJob)
+
+	assert.NotEmpty(t, err)
```
Comment (ccremer) [inline stardog/test/postinstall_test.go:24]: The `assert.NoError(t, err)` is better suited for testing errors. please use that.

File: stardog/test/postinstall_test.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+	batchv1 "k8s.io/api/batch/v1"
+
+	"github.com/gruntwork-io/terratest/modules/helm"
+	"github.com/stretchr/testify/assert"
+)
+
+var tplPostInstallJob = []string{"templates/post-install-job.yaml"}
+
+func Test_PostInstallJob_Should_Not_Exist(t *testing.T) {
```
Comment (ccremer) [inline stardog/test/postinstall_test.go:14]: A more fitting title: `Test_PostInstallJob_GivenReplicaCount_WhenZero_ThenDoNotRenderJob`

File: stardog/test/postinstall_test.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+		},
+	}
+
+	output, err := helm.RenderTemplateE(t, options, helmChartPath, releaseName, tplPostInstallJob)
+
+	assert.NotEmpty(t, err)
+	assert.Equal(t, "Error: could not find template templates/post-install-job.yaml in chart", output)
+}
+
+func Test_PostInstallJob_Should_Exist(t *testing.T) {
```
Comment (ccremer) [inline stardog/test/postinstall_test.go:28]: A more fitting title: `Test_PostInstallJob_GivenReplicaCount_WhenGreaterThanZero_ThenRenderJob`

File: stardog/test/postinstall_test.go (hunk 1 above)
Comment (srueg) [reply stardog/test/postinstall_test.go:24]: It's the exact opposite: `assert.Error(t, err)`

Comment (ccremer) [reply stardog/test/postinstall_test.go:24]: riiight, mixed that up :)

**Comment by chamer81** [comment]
Replaced by #187

**Pull request closed without merging**